### PR TITLE
Included Compat for backwards compatibility

### PR DESCRIPTION
**Pull request by hustf**
Refer https://github.com/JuliaLang/IniFile.jl/issues/13

### CODE REVIEW

**Comment by hustf** [comment]
Also added appveyor.yml file.

File: src/IniFile.jl [hunk 1]
```diff
@@ -21,11 +22,11 @@ export Inifile,
        sections,
        show
 
-typealias INIVAL Union{AbstractString,Number,Bool,Void}
+@compat typealias INIVAL Union{AbstractString,Number,Bool,Void}
 typealias HTSS Dict{AbstractString,INIVAL}
 
 type Inifile
-    sections::Dict{AbstractString,HTSS}
+@compat    sections::Dict{AbstractString,HTSS}
```
Comment (nalimilan) [inline src/IniFile.jl]: Too many spaces here. Actually, I think you should align `@compat` to the line below using 4 spaces.

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 language: julia
 os:
-  - linux
```
Comment (nalimilan) [inline .travis.yml]: Any reason you change the indentation here? At least it's not consistent with `email` below.

**Comment by nalimilan** [comment]
Thanks for adding the AppVeyor tests. I trust you this passes the tests on both 0.3 and 0.4?

File: src/IniFile.jl (hunk 1 above)
Comment (tkelman) [inline src/IniFile.jl]: the compat macro is only needed for dict construction, not the type

**Comment by hustf** [comment]
Thanks for the feedback; making sloppy contributions is often worse than doing none.

@nalimilan,  Previous experience makes me hesitate to promise this is going to work; I do promise that it works where I have tested.  I´ll check the program again on 0.4 Windows 64 after the next changes, and also on Win32. Having checks made automatically on commits and on releases on all platforms is going to further Julia development significantly I hope. But I have not registered my on Appveyor, and I have never tried this before. I suppose this is better done by a package owner.

Regarding travis.yml. intendation, I noticed it was different in Compat, which do work in contrast to this package. Again, I do not really know the system. There is a template package somewhere which would be a better source.

@tkelman,  I didn´t expect inserting Compat on line 29 would help, but it seemed to fix some error message. It was thrown in as a temporary debug check. Perhaps that was quite unrelated; I will go a little bit deeper into it and tidy up the indendtation. I notice a lot of package code throws @Compat into the middle of lines, and I suppose that is just ineffective.

**Comment by tkelman** [comment]
We're glad to have new contributors as long as we can walk through the parts you're unfamiliar with and get you up to speed. We all had to start somewhere.

`@compat` (best to escape macros with backticks to avoid notifying unrelated github users) works either at the middle of a line, localized to the specific syntax construction where backwards-compatibility is needed, or on an entire line since the macro is recursive. It's a matter of style, I prefer the localized version but others may rather just use a single instance for an entire long complicated line of code.

**Comment by hustf** [comment]
@tkelman, thanks for the encouragment. This is not the place for that discussion, but I see a lot of warning messages regarding redefined functions when packages are using Compat. Why is this / where is it discussed?

**Comment by tkelman** [comment]
The julia-users mailing list is probably the right place to ask questions if you're not sure where they should go. Post the warnings you're seeing there along with (a link to a gist with) `versioninfo()` and `Pkg.status()`, if there's a bug then someone will say to open an issue and add cross-reference links between the issue and the mailing list thread.

Regarding CI, I just enabled Travis here. We can also enable AppVeyor, all it needs is someone with collaborator status to volunteer to have it running on their account. I would do it but my AppVeyor account is already pretty crowded with other packages and the free plan only has one concurrent build per account. Any other volunteers?

**Comment by hustf** [comment]
@nalimilan, the indentation switched back to the norm as given by Pkg.generate("FooBar","MIT")
Appveyor.yml is unchanged from Compat, which is dissimilar to the above norm.
Testen on Windows 32 bit 3.10 and 4.0, 64 bit 3.11  and 4.0.
There is a delay of several minutes at exit of Pkg.test() on 3.10. But the delay occurs after success is confirmed by Pkg.test.

**Comment by nalimilan** [comment]
@hustf Thanks, looks good.

@tkelman Shouldn't we add this to the Julia ("StefanKarpinski") account on AppVeyor, since the package is owned by JuliaLang?

**Comment by tkelman** [comment]
I'd rather avoid adding any packages to Stefan's account, since those packages would then share queue space with base Julia commits and PR builds. Anyone with contributor access should be able to create a free account on AppVeyor and it would do for this low-traffic package.

**Comment by nalimilan** [comment]
Here you go: https://ci.appveyor.com/project/nalimilan/inifile-jl

**Comment by nalimilan** [comment]
@tkelman BTW, is this good to merge? I don't know anything about AppVeyor config files.

**Comment by tkelman** [comment]
Thanks @nalimilan! It looks like I'll have to go in and enable the webhook manually. If you go to the settings of the inifile-jl project, can you copy-paste the Webhook URL?

**Comment by nalimilan** [comment]
It's https://ci.appveyor.com/api/github/webhook?id=i2jhppaesdtyxs3y

**Comment by tkelman** [comment]
Great, I enabled the webhook. To see it in action let's have @hustf push one more commit here. How about adding 0.4 to the appveyor matrix with
```
  - JULIAVERSION: "julialang/bin/winnt/x86/0.4/julia-0.4-latest-win32.exe"
  - JULIAVERSION: "julialang/bin/winnt/x64/0.4/julia-0.4-latest-win64.exe"
```
between the existing 0.3 and nightly?

**Comment by hustf** [comment]
Ok,with your permission considering 
http://docs.julialang.org/en/release-0.4/manual/packages/?highlight=package%20development
```` If you don’t edit the dash cruft you are suggesting that you support both the development and stable versions of the same version number! That would be madness. ´´´´

**Comment by tkelman** [comment]
If there's a question there, please phrase it differently. That section is referring to trailing dashes (aka prereleases) on version numbers in `REQUIRE` file, which you are correctly removing here.

**Comment by hustf** [comment]
I should have put in a smiley there I suppose. Sorry for that, but it seems this is working now.

**Comment by nalimilan** [comment]
Thanks! This kind of compatibility fixes + tests is really appreciated.

**Pull request merged**